docs/guides/how-to-write-a-gadget.md

```diff
@@ -0,0 +1,141 @@
+# How to write a Gadget?
+
+So you want to write a Gadget for Inspektor Gadget? Great!
+
+A gadget consists of a __Gadget Descriptor__ that provides metadata about your gadget and a __Gadget__ implementation
+(sometimes called tracer in our codebase).
+
+## Gadget Descriptor
+
+The first step to writing a new gadget is providing a _Gadget Descriptor_. Simply create a new struct that implements
+the `gadgets.GadgetDesc` interface, providing all required information, including a name, description, category and so
+on. In our example, we call this struct like the interface, `GadgetDesc`. If you want your gadget to return records of
+a specific known type, for example system events as they come in, you can define a type for it (we call it
+EventDataType in our example) and return a pointer to an empty version of it in the `EventPrototype()` methods and
+initialize a Parser inside the `Parser()` function. This typically looks like this:
+
+```go
+func (g *GadgetDesc) EventPrototype() {
+	return &EventDataType{}
+}
+
+func (g *GadgetDesc) Parser() {
+	return parser.NewParser[EventDataType](nil)
+}
+```
+
+If you want support for a human-readable output in the form of text columns later on,
+you should annotate the fields of your data struct like explained in our
+[Columns library documentation](https://pkg.go.dev/github.com/inspektor-gadget/inspektor-gadget@v0.13.0/pkg/columns).
+The annotation works similar to the JSON tags for struct members that you probably already know.
+In that case, you should instantiate a columns helper in the `Parser()` function like so:
+
+```go
+func (g *GadgetDesc) Parser() {
+	return parser.NewParser[EventDataType](columns.MustCreateColumns[EventDataType]())
+}
+```
```

Comment:
I was wondering if the gadget should have a method `GetColumns()` instead of `Parser()` and the caller (`buildCommandFromGadget` in our case) is in charge of creating the parser as it already knows the type from `EventPrototype()`. In this way, the gadget developer doesn't have to know about the `Parser` but just to return what `columns.MustCreateColumns[EventDataType]()` (`types.GetColumns()` in our case) returns if they want to support "human-readable" output or return `nil` otherwise. WDYT?

Reply:
I'd really like that, but afaiu it's not as simple as that, because types that use generics have to be known at compile time and this would make it more runtime-ish. But maybe I fail to see how it can be done right now - will definitely look more into it.